thread-utils.c

```diff
@@ -18,6 +18,67 @@
 #  endif
 #endif
 
+#ifdef GIT_WINDOWS_NATIVE
+/// defines the GetLogicalProcessorInformationEx function
+typedef BOOL (*WINAPI glpie_t)(
+	LOGICAL_PROCESSOR_RELATIONSHIP,
+	PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX,
+	PDWORD);
+
+// Classic way to get number of cores in windows.
+static size_t get_windows_compatible_n_cores() {
```

Comment:
Let's not add an extra function here. Instead, let's declare a single `win32_online_cpus(void)` function in a header file in `compat/win32/` and implement it in a `.c` file. Or put them into `compat/mingw.h`/`compat/mingw.c`.